### PR TITLE
DonateHelpPage model, factory, and tests

### PR DESCRIPTION
**Pull request by danielfmiranda**
Related Ticket: #11172

Link to review page: https://donate-foundation-s-11177-dona-5stlkm.mofostaging.net/en/help/

# Description

Originally, there were only two page models related to donations on the foundation stack. These were:
- `BaseDonationPage` - Which serves as the base page model, in case we need to add any shared fields/functionality across Donation related models. (currently, there are none so its pretty much blank.)
- `DonationLandingPage` - Which serves as the landing page for the donation related pages.

These both lived in a file `models.py`, which goes against what we do for other models in our codebase. 

If you take a look at the folder `wagtailpages` for example, you will see that `models.py` imports all page models, which all live under their respective folders/files. (for example `blog/blog_page.py`)

Since we are so early into the development of the donate side of the foundation site, I thought that this PR would be the best time to do some refactoring so the donate models can fall under the same convention. 

That is the reason that alongside the `DonateHelpPage` model changes, you will also see the following additional file changes:

- `pagemodels/__init__.py`
- `pagemodels/base_page.py`
- `pagemodels/landing_page.py`

The code related to these page models has already been approved in [this PR](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/10401), so I do not believe there is a need to look too into these.

**That being said, this PR introduces the following DonateHelpPage related updates:**

- A new model `DonateHelpPage` which at the moment only has one  `StreamField` field `body`. The streamblocks available for use are the same as `CampaignPage` per request of design. (They both use our pre-defined list called `base_fields`)
- A template for the new model, which at the moment only renders the page's title, as well as placeholders for "Intro", "Form", and "Body"
- A new factory for the model, and an update to existing factories so new db's will include a help page titled "Donate Help"
- A basic test suite for the new model, as well as one small update to the `DonateLandingPage` tests, confirming that it can create a `DonateHelpPage`.

# Screenshots
Screenshot 2023-09-27 at 15-40-58 Donate Help

# Steps to test

1. Checkout this branch
2. Run `inv new-db`
3. Log into the CMS
4. Alongside "Homepage" and "Mozilla Festival", there should be a page titled "Donate Now" with a subpage "Donate Help".
5. Click into "Donate Help" and take a look around, it should only have two fields "title" and "body".
6. Visit the page on the front end, it should look like the screenshot above.
7. If everything is working as expected, testing is complete!

### CODE REVIEW

**Review by mmmavis — APPROVED** [review]
Awesome! Thanks for the clear review instructions as well as breaking the existing code into separate files.

File: network-api/networkapi/donate/pagemodels/help_page.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    translatable_fields = [
+        # Promote tab fields
+        SynchronizedField("slug"),
+        TranslatableField("seo_title"),
+        SynchronizedField("show_in_menus"),
+        TranslatableField("search_description"),
+        SynchronizedField("search_image"),
+        # Content tab fields
+        SynchronizedField("body"),
```
Comment (jhonatan-lopes) [inline network-api/networkapi/donate/pagemodels/help_page.py:33]: Is this correct? Do we want to show the same `body` for all translated pages? Shouldn't this be a `TranslatableField`?

File: network-api/networkapi/donate/factory.py [hunk 2]
```diff
@@ -18,6 +18,13 @@ class Meta:
     featured_image = SubFactory(ImageFactory)
 
 
+class DonateHelpPageFactory(PageFactory):
+    class Meta:
+        model = DonateHelpPage
+
+    title = Faker("text", max_nb_chars=140)
```
Comment (jhonatan-lopes) [inline network-api/networkapi/donate/factory.py:25]: Just FYI, I'm usually using `Faker("sentence")`, since it's usually better for a phrase.

**Review by jhonatan-lopes — COMMENTED** [review]
Looks good, I just have one question for a translated field...

File: network-api/networkapi/donate/pagemodels/help_page.py (hunk 1 above)
Comment (danielfmiranda) [reply network-api/networkapi/donate/pagemodels/help_page.py:33]: Hey @jhonatan-lopes! Great point, after taking a look at the rest of our codebase it appears that I put this here by mistake. All of the other "body" fields are set to be a `TranslatableField` like you mentioned. I have updated this file to do the same! 

Thanks for taking a look 👍

**Comment by danielfmiranda** [comment]
Hey @jhonatan-lopes thanks for taking a look and the great feedback! I have updated the code accordingly and re-requested review for whenever you find a free moment 🙌

**Review by jhonatan-lopes — APPROVED** [review]
LGTM :+1:

**Pull request merged**